cmd/whclient/main.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"io"
+	"net"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/taskcluster/taskcluster-client-go"
+	"github.com/taskcluster/taskcluster-client-go/auth"
+	"github.com/taskcluster/webhooktunnel/util"
+	"github.com/taskcluster/webhooktunnel/whclient"
+)
+
+// whclient <clientID> <accessToken> <targetPort>
+const usage = `whcli <clientID> <accessToken> <targetPort>`
+
+func main() {
+	log.SetOutput(os.Stdout)
+	log.SetFormatter(&log.JSONFormatter{})
+
+	if len(os.Args) != 4 {
+		log.Fatal(usage)
+	}
+
+	clientID, accessToken := os.Args[1], os.Args[2]
+	targetPort, err := strconv.Atoi(os.Args[3])
+	if err != nil || targetPort < 0 || targetPort > 65535 {
+		log.Fatal(usage)
+	}
+
+	configurer := func() (whclient.Config, error) {
+		creds := &tcclient.Credentials{
+			ClientID:    clientID,
+			AccessToken: accessToken,
+		}
+		myAuth := auth.New(creds)
+		whtResponse, err := myAuth.WebhooktunnelToken()
+		if err != nil {
+			return whclient.Config{}, err
+		}
+		proxyURL := util.MakeWsURL(whtResponse.ProxyURL)
+		return whclient.Config{
+			ID:        whtResponse.TunnelID,
+			Token:     whtResponse.Token,
+			ProxyAddr: proxyURL,
+			UseDomain: true,
+		}, nil
+	}
+
+	client, err := whclient.New(configurer)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.WithFields(log.Fields{"url": client.URL()}).Info("connected to proxy")
+
+	waitTime := 5 * time.Millisecond
```

Comment:
Maybe add a comment over this line, saying what it is that is going on here...

Maybe handle sigint and sigterm? to gracefully terminate... ie. if we get a sigint or a sigterm we stop accepting new connections... and let existing connections finish, if they don't finish within say 5 seconds we kill them and then we terminate the program...